go/arrow/array/data_test.go

```diff
@@ -49,3 +49,46 @@ func TestDataReset(t *testing.T) {
 		data.Reset(&arrow.Int64Type{}, 5, data.Buffers(), nil, 1, 2)
 	}
 }
+
+func TestSizeInBytes(t *testing.T) {
+
+	var buffers1 = make([]*memory.Buffer, 0, 3)
+
+	for i := 0; i < cap(buffers1); i++ {
+		buffers1 = append(buffers1, memory.NewBufferBytes([]byte("15-bytes-buffer")))
+	}
+
+	// test 1: data with buffers only
+	data := NewData(&arrow.StringType{}, 10, buffers1, nil, 0, 0)
+	expectedSize := uint64(45)
+	if actualSize := data.SizeInBytes(); actualSize != expectedSize {
+		t.Errorf("expected size %d, got %d", expectedSize, actualSize)
+	}
```

Comment:
can you wrap these with `t.Run(<test name>, func(t *testing.T) {` so they show up as individual tests which can be run/debugged independantly? It makes it easier to find and diagnose if a test fails.